### PR TITLE
support OpenThread time synchronization service

### PR DESCRIPTION
**Pull request by chshu**
Three new properties are introduced:
 * TimeSync:ThreadNetworkTime
 * TimeSync:Period
 * TimeSync:XtalThreshold

Corresponding OpenThread PR: [PR#2618](https://github.com/openthread/openthread/pull/2618)

### CODE REVIEW

File: src/ncp-spinel/SpinelNCPInstance.cpp [hunk 1]
```diff
@@ -747,6 +747,67 @@ unpack_dataset(const uint8_t *data_in, spinel_size_t data_len, boost::any &value
...
+		if (as_val_map) {
+			entry.clear();
+			entry[kWPANTUNDValueMapKey_TimeSync_Time] = time;
+			entry[kWPANTUNDValueMapKey_TimeSync_Status] = timeSyncStatus;
+			result_as_val_map.push_back(entry);
+		} else {
+			char c_string[500];
+			int index;
+
+			index = snprintf(c_string, sizeof(c_string), "ThreadNetworkTime: %lu, TimeSyncStatus:%d", time, timeSyncStatus);
```
Comment (bukepo) [inline src/ncp-spinel/SpinelNCPInstance.cpp]: use PRIu64 instead of %lu to support 32bit system.

Comment (chshu) [reply src/ncp-spinel/SpinelNCPInstance.cpp]: Fixed, thanks.

**Comment by chshu** [comment]
@abtink The related OT [PR#2618](https://github.com/openthread/openthread/pull/2618) was merged, and this PR has been rebased to latest master, please help to review, thanks.

File: src/ncp-spinel/SpinelNCPInstance.cpp [hunk 2]
```diff
@@ -542,6 +543,13 @@ SpinelNCPInstance::get_supported_property_keys()const
 		properties.insert(kWPANTUNDProperty_NestLabs_LegacyMeshLocalPrefix);
 	}
 
+        if (mCapabilities.count(SPINEL_CAP_TIME_SYNC))
```
Comment (abtink) [inline src/ncp-spinel/SpinelNCPInstance.cpp]: nit: extra space before `if`?

Comment (chshu) [reply src/ncp-spinel/SpinelNCPInstance.cpp]: Updated, thanks.

**Pull request merged**